tuning_docs/tensile_tuning.tex

```diff
@@ -579,61 +637,97 @@ \subsubsection{The Tensile Logic File}
+\noindent
+Note that as of now, rocBLAS does not hold the ld information in the final library, even if we define them in the configure file.
+
+\bigskip
+\noindent
+
+\textbf{\large{Naming convention:}}
+
+\label{sec:naming}
+\noindent
+Tensile generates one library logic file for each problem type. They are named as follows:
+\begin{center}
+	$\langle architecture\_name \rangle \_\langle Contraction\_Type \rangle\_ \langle Data\_Type \rangle$
+\end{center}
+
+\noindent
+$\langle architecture\_name \rangle$ refers the GPU architecture, for example for Vega 20, the architecture name is vega20. $\langle Contraction\_Type \rangle$ refers to the Einstein tensor convention as discussed in section \ref{sec:einstein}. $\langle Data\_Type \rangle$ is based on the ProblemType in the configuration file. For example, SB refers to a Batched Single precision DataType. HSS\_BH means Half-precision for DataType (A/B), Single-precision for DestDataType (C/D), and Single-precision for ComputeDataType. Notice also that the exact same naming convention will be used in the rocBLAS library.
+
+\section{Integrating new libraries in rocBLAS}
+Once we have library logic files for new sizes, we need to merge and update rocBLAS libraries. This a two-fold process: 
+\begin{itemize}
+\item First, we need to confirm that the new logic file is generating the correct results and the performance of the new libraries is more that the old library. We use rocBLAS-bench tool for this part. See the next section. 
+\item Merging the new logic file into rocBLAS using the merge tool (merge.py in Tensile). See section \ref{sec:merging}.
+\end{itemize}
+
+
+
+\subsection{rocBLAS-bench tool}
+
+\label{sec:bench}
+
+\noindent
+rocBLAS-bench is a tool used to perform benchmarks on the problem sizes which are being measured and to verify the correctness of the new libraries. While it is designed to measure performance of the full set of rocBLAS library functions, our interest is in the level 3 gemm functions which---in rocBLAS---are calls into the Tensile library. 
 
-rocblas-bench is a tool used to perform benchmarks on the problem sizes which are being measured. While it is designed to measure performance of the full set of rocBLAS library functions, our interest is in the level 3 gemm functions which---in rocBLAS---are calls into the Tensile library.
 
 The git repository for rocBLAS is found at \url{https://github.com/ROCmSoftwarePlatform/rocBLAS}. Either the develop or master branch version may be used for testing.
 
 \begin{lstlisting}[language=bash]
-git clone -b master|develop https://github.com/ROCmSoftwarePlatform/rocBLAS.git
+$ git clone -b master|develop https://github.com/ROCmSoftwarePlatform/rocBLAS.git
 \end{lstlisting}
 
-\noindent
-The library can be installed using the included installation script:
+To use rocBLAS-bench we need to install it locally. The library can be installed using the included installation script (the installation may last quite some time.):
```

Comment:
Not strictly true; the rocblas-bench (and other clients) can be built using the installed version of rocBLAS without having to first rebuild rocBLAS. I think the option is --clients-only.